src/tests/logic/calculate.spec.js

```diff
@@ -0,0 +1,58 @@
+import calculate from '../../logic/calculate';
+
+const data = {
+  total: null,
+  next: null,
+  operation: null,
+};
+
+describe('calculate', () => {
+  it('returns null object if button is AC', () => {
+    const newData = calculate(data, 'AC');
+
+    expect(newData).toEqual(data);
+  });
+
+  it('returns valid object when button is - and opeation is null', () => {
+    const newData = calculate(data, '-');
+
+    expect(newData).toEqual({ next: '0-', operation: '-' });
+  });
+
+  it('returns null object when operation ends with n', () => {
+    const newData = calculate(
+      {
+        total: null,
+        next: null,
+        operation: 'n',
+      },
+      '-',
+    );
+
+    expect(newData).toEqual({ next: null, operation: null });
+  });
+
+  it('returns valid object when button is %', () => {
+    const newData = calculate(data, '%');
```

Comment:
- [ ] We need to test if the modifier operator is working as expected, like:
`Before: 5`
`After: 0.05`